src/main/java/fr/minuskube/inv/util/ItemBuilder.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2018-2020 Isaac Montagne
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package fr.minuskube.inv.util;
+
+import fr.minuskube.inv.ClickableItem;
+import fr.minuskube.inv.ItemClickData;
+import org.bukkit.Color;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.LeatherArmorMeta;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * This is a simple class that simplifies the creation of ItemStacks
+ */
+public class ItemBuilder {
```

Comment:
All of the things like parameters, classes, fields(stack and meta) should be final(un-changable) because of the safety. Especially paramters.

Reply:
I think fields (and classes in some cases) can be final, but please don't make parameters and local variables final here, I think it's not really worth the readability cost.

But I'll make a Code Style Guide ASAP to clarify these things.

Reply:
for parameters, we should know information about "the parameters won't change" for security.

Reply:
In Java versions below Java 12 the final modifier effectively does nothing on fields. And on classes, it just prevents extending them but Java Reflection come again to ruin the day in that perspective too